test/functional/combine_logs.py

```diff
@@ -73,10 +73,12 @@ def get_log_events(source, logfile):
                 # if this line has a timestamp, it's the start of a new log event.
                 time_match = TIMESTAMP_PATTERN.match(line)
                 if time_match:
-                    if event:
-                        yield LogEvent(timestamp=timestamp, source=source, event=event.rstrip())
-                    event = line
-                    timestamp = time_match.group()
+                    if event or timestamp:
```

Comment:
Could keep this as `if event` to not show the (initial) empty log message. That's a matter of taste, I guess.